common/etc/nginx/templates/gateway/cors.conf.template

```diff
@@ -0,0 +1,31 @@
+set $request_cors "${request_method}_${CORS_ENABLED}";
+
+if ($request_cors = "OPTIONS_1") {
+    add_header 'Access-Control-Allow-Origin' '*';
+    add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+    #
+    # Custom headers and headers various browsers *should* be OK with but aren't
+    #
+    add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range';
+    #
+    # Tell client that this pre-flight info is valid for 20 days
+    #
+    add_header 'Access-Control-Max-Age' 1728000;
+    add_header 'Content-Type' 'text/plain; charset=utf-8';
+    add_header 'Content-Length' 0;
+    return 204;
+}
```

Comment:
For the `OPTIONS` if, it seems to fall under the category of "good ifs":
> There are cases where you simply cannot avoid using an if, for example, if you need to test a variable which has no equivalent directive.

As for the other ifs, I believe they are ok due to there being no ifs after them. However, I have no idea how you would be able to do the same operation with map. Ideally, I wanted to conditionally template in the cors.conf include, but that seemed a bit too clunky. I guess I could conditional set an env var with the whole include statement in `docker-entrypoint.sh`.

Reply:
Yeah this certainly is the best for understandibility.  I wouldn't add more moving parts if it's testing ok.   I'm going to play around with making it work with map because I'm curious now but let's not let it hold up the PR.